src/Protocol/ActivityPub/Transmitter.php

```diff
@@ -0,0 +1,910 @@
+
+		return $inboxes;
+	}
+
+	/**
+	 * @brief Returns the activity type of a given item
+	 *
+	 * @param array $item
+	 *
+	 * @return activity type
+	 */
+	private static function getTypeOfItem($item)
+	{
+		if ($item['verb'] == ACTIVITY_POST) {
+			if ($item['created'] == $item['edited']) {
+				$type = 'Create';
+			} else {
+				$type = 'Update';
+			}
+		} elseif ($item['verb'] == ACTIVITY_LIKE) {
+			$type = 'Like';
+		} elseif ($item['verb'] == ACTIVITY_DISLIKE) {
+			$type = 'Dislike';
+		} elseif ($item['verb'] == ACTIVITY_ATTEND) {
+			$type = 'Accept';
+		} elseif ($item['verb'] == ACTIVITY_ATTENDNO) {
+			$type = 'Reject';
+		} elseif ($item['verb'] == ACTIVITY_ATTENDMAYBE) {
+			$type = 'TentativeAccept';
+		} else {
+			$type = '';
+		}
+
+		return $type;
+	}
+
+	/**
+	 * @brief Creates an activity array for a given item id
+	 *
+	 * @param integer $item_id
+	 * @param boolean $object_mode Is the activity item is used inside another object?
+	 *
+	 * @return array of activity
+	 */
+	public static function createActivityFromItem($item_id, $object_mode = false)
+	{
+		$item = Item::selectFirst([], ['id' => $item_id, 'parent-network' => Protocol::NATIVE_SUPPORT]);
+
+		if (!DBA::isResult($item)) {
+			return false;
+		}
+
+		$condition = ['item-uri' => $item['uri'], 'protocol' => Conversation::PARCEL_ACTIVITYPUB];
+		$conversation = DBA::selectFirst('conversation', ['source'], $condition);
+		if (DBA::isResult($conversation)) {
+			$data = json_decode($conversation['source']);
+			if (!empty($data)) {
+				return $data;
+			}
+		}
+
+		$type = self::getTypeOfItem($item);
+
+		if (!$object_mode) {
+			$data = ['@context' => ActivityPub::CONTEXT];
+
+			if ($item['deleted'] && ($item['gravity'] == GRAVITY_ACTIVITY)) {
+				$type = 'Undo';
+			} elseif ($item['deleted']) {
+				$type = 'Delete';
+			}
+		} else {
+			$data = [];
+		}
+
+		$data['id'] = $item['uri'] . '#' . $type;
+		$data['type'] = $type;
+		$data['actor'] = $item['author-link'];
+
+		$data['published'] = DateTimeFormat::utc($item["created"]."+00:00", DateTimeFormat::ATOM);
+
+		if ($item["created"] != $item["edited"]) {
+			$data['updated'] = DateTimeFormat::utc($item["edited"]."+00:00", DateTimeFormat::ATOM);
+		}
+
+		$data['context'] = self::fetchContextURLForItem($item);
+
+		$data = array_merge($data, self::createPermissionBlockForItem($item));
+
+		if (in_array($data['type'], ['Create', 'Update', 'Announce', 'Delete'])) {
+			$data['object'] = self::createNote($item);
+		} elseif ($data['type'] == 'Undo') {
+			$data['object'] = self::createActivityFromItem($item_id, true);
+		} else {
+			$data['object'] = $item['thr-parent'];
+		}
+
+		$owner = User::getOwnerDataById($item['uid']);
+
+		if (!$object_mode) {
+			return LDSignature::sign($data, $owner);
+		} else {
+			return $data;
+		}
+
+		/// @todo Create "conversation" entry
+	}
+
+	/**
+	 * @brief Creates an object array for a given item id
+	 *
+	 * @param integer $item_id
+	 *
+	 * @return object array
```

Comment:
if the return variable could be both (either object or array) please use `@return object|array`

Reply:
Or is the return variable an `array of the object` ?